next-app/src/components/features/shows/AllShowsShowList/AllShowsShowList.tsx

```diff
@@ -0,0 +1,23 @@
+import ShowList from "@/components/shared/ShowList/ShowList";
+import { getShows, getShowsKey } from "@/fetchers/shows";
+import { HStack, Spinner, VStack } from "@chakra-ui/react";
+import React from "react";
+import useSWR from "swr";
+
+export const AllShowsShowList = () => {
+  const { data, isLoading, error } = useSWR(getShowsKey(), getShows);
+
+  if (isLoading) {
+    return (
+      <VStack py={16}>
+        <Spinner size="xl" />
+      </VStack>
+    );
+  }
+
+  if (error) {
+    return <div>Error: {error.message}</div>;
+  }
+
+  return <ShowList shows={data!.shows} />;
```

Comment:
Okej način za natjerati typescript da vjeruje da ce data nece nikad biti undefined ali to i nije istina
Ja bi radje ovo reorganizirao na način:
- provjeru za error stavio prvu
- updejtao provjeru za loding na nacin (isLoading || !data)

Na taj način je loading dokle god ne dodje neki data, a opet ukoliko prvo sleti error neće biti infinite load, a time nam i ne treba ovaj ! operator :)